### PR TITLE
285 rfm model

### PR DESCRIPTION
**Pull request by bdeck8317**
Moved rfm functions to 
`src/server/rfm/`

### CODE REVIEW

File: src/server/rfm/rfm_funcs/amount_bins.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+
+    # human_amount_bins = big_donations[['Amount']].quantile([.2,.4,.6,.8]).reset_index()
+
+
+    # human_amount_bins = bins[['Amount']].to_numpy().flatten()
+    # lower_bin = 0
+    # upper_bin = np.inf
+    # human_amount_bins = np.insert(bins, 0, lower_bin)
+    # human_amount_bins = np.insert(bins, 5, upper_bin)
```
Comment (c-simpson) [inline src/server/rfm/rfm_funcs/amount_bins.py:22]: Any reason not to remove the commented-out code?

File: src/server/rfm/rfm_funcs/create_bins.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    import numpy as np
+    import jenkspy
+    from datetime import datetime, date
+    import os
+
+
+
+    ####
+    # read in csv
+    donations_df = pd.read_csv('csv')
```
Comment (c-simpson) [inline src/server/rfm/rfm_funcs/create_bins.py:17]: Aren't we getting data from tables (via `pull_donations_for_rfm()`), not from files?

File: src/server/rfm/rfm_funcs/create_bins.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    donations_df['Close_Date'] =pd.to_datetime(donations_2021['Close_Date']).dt.date
```
Comment (c-simpson) [inline src/server/rfm/rfm_funcs/create_bins.py:19]: Could we use a variable name more general than `donations_2021`?

File: src/server/rfm/rfm_funcs/create_bins.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        def checkIfDuplicates(listOfElems):
+        ''' Check if given list contains any duplicates '''
+        for elem in listOfElems:
+            if listOfElems.count(elem) > 1:
+                return True
+        return False
+
+        duplicats_bool = checkIfDuplicates(jenks_frequency_bins)
+        if duplicates_bool  == True:
+            final_frequency_bins = human_frequency_bins
```
Comment (c-simpson) [inline src/server/rfm/rfm_funcs/create_bins.py:42]: Are these supposed to be the same variable? If not, maybe change one to be less similar.

File: src/server/rfm/rfm_funcs/create_scores.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    '''
+
+    import pandas as pd
+    import numpy as np
+    from datetime import datetime, date
+    from collections import Counter
+    from admin_api import read_rfm_edges, pull_donations_for_rfm
+
+
+    df = pd.read_csv(path_to_csv)
```
Comment (c-simpson) [inline src/server/rfm/rfm_funcs/create_scores.py:13]: CSV file?

File: src/server/rfm/rfm_funcs/create_scores.py [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    frequency_bins  =  list(read_rfm_edges('f').values())             #       imported from table
+
+    monetary_labels = [ 1,2,3,4,5]
+    monetary_bins =   list(read_rfm_edges('m').values())              #       imported from table
+
+
+    ########################## recency #########################################
+
+
+    donations_2021 = df
```
Comment (c-simpson) [inline src/server/rfm/rfm_funcs/create_scores.py:34]: **donations_past_year** maybe?

File: src/server/rfm/rfm_funcs/create_scores.py [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    donations_2021['close_date'] =pd.to_datetime(donations_2021['close_date']).dt.date
+
+        # calculate date difference between input date and
```
Comment (c-simpson) [inline src/server/rfm/rfm_funcs/create_scores.py:37]: input date and.. ?

File: src/server/rfm/rfm_funcs/create_scores.py [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+    from rfm_functions import date_difference
+    days = []
+    for ii in donations_2021['close_date']:
+        days.append(date_difference(ii, str(query_date)))
+    donations_2021['days_since'] = days
+
+    grouped_2021 = donations_2021.groupby('_id').agg({'days_since': ['min']}).reset_index()
```
Comment (c-simpson) [inline src/server/rfm/rfm_funcs/create_scores.py:44]: Same concern about 2021

File: src/server/rfm/rfm_funcs/rfm_functions.py [hunk 9]
```diff
@@ -0,0 +1,44 @@
+# rfm_funcs
+
+### Date difference for recency score calculation
+
+# get date difference
+def date_difference(my_date, query_date):
+    from datetime import datetime, date
```
Comment (c-simpson) [inline src/server/rfm/rfm_funcs/rfm_functions.py:7]: Can you add a docstring - what does the diff represent  - days?

**Review by c-simpson — CHANGES_REQUESTED** [review]
What do we do with the giant html files?
Can you add some comments that explain how to run your code?  Where to start?
What do you do with the scores when you are finished?
(See other comments below)

**Comment by bdeck8317** [comment]
I will make all the above changes and create a new pull request!

**Comment by bdeck8317** [comment]
All requested changes have been made. Removed the html doc and created a readme.md for process of running the functions.

**Pull request closed without merging**